src/ClientOptionsTrait.php

```diff
@@ -178,6 +177,11 @@ private function buildClientOptions(array $options)
             $apiEndpoint = self::determineMtlsEndpoint($options['apiEndpoint']);
         }
 
+        // If the user has not supplied a universe domain, use the environment variable if set.
+        // Otherwise, use the default ("googleapis.com").
+        $options['universeDomain'] ??= getenv('GOOGLE_CLOUD_UNIVERSE_DOMAIN')
```

Comment:
TIL this, `??=`, null coalescing variant! I love these types of quality of life features PHP has.

Reply:
I love it too! I wanted to write an RFC for it and then found out it was already implemented :D